### PR TITLE
Performance adjustment for tankless dhw

### LINKED ISSUE #31: Add documentation of performanceAdjustment for tankless water heaters
- [x] baseBuilding.systems.waterHeatingSystems
- [x] improvedBuilding.systems.waterHeatingSystems
- [x] automatedMeasures.newWaterHeatingSystem

### PR DESCRIPTION
**Pull request by kloconnell**

### CODE REVIEW

**Comment by kloconnell** [comment]
@joshuamzm this PR includes some small changes to include documentation of `performanceAdjustment`, which affects instantaneous (i.e. tankless) water heaters. I've added the necessary changes in hpxml-gen workbook, but the code repo will need updating too. Can you merge this PR when the code repo is ready? This is not super urgent.

**Review by joshuamzm — APPROVED** [review]

**Comment by joshuamzm** [comment]
@kellyatradiantlabs All changes in this PR look great. I created a ticket to implement those changes in the API here: https://github.com/RadiantLabs/core-modeling-api/issues/200. Until then, we'll keep this PR open.

File: docs/source/api.rst [hunk 1]
```diff
@@ -727,7 +727,8 @@ Property                                     Type     Units
 ``heatCapacityBtuPerHour``                   float    Btu/hr                      >0              no                              autosized by modeling engine if undefined
 ``energyFactor`` or ``uniformEnergyFactor``  float    fraction                    <1              no        BSA
 ``hotWaterTemperature``                      float    F                           >0              no        125 
-===========================================  =======  ==========================  ==============  ========  ====================  ============================================== 
+``performanceAdjustment``                    float    fraction                    see [#]_        no        see [#]_              Multiplier on efficiency, typically to account for cycling
```
Comment (joshuamzm) [inline docs/source/api.rst]: @kellyatradiantlabs Is this decimal number greater or equal or strictly greater than zero?

Comment (kloconnell) [reply docs/source/api.rst]: Good question. OpenStudio does not specify a constraint on this value in their documentation, but I think we should implement a 0-1 constraint. I'll revise in our docs accordingly.

**Comment by joshuamzm** [comment]
@kellyatradiantlabs Support for `performanceAdjustment` has been added in https://github.com/RadiantLabs/core-modeling-api/issues/200 and included in the release of V6. I updated this PR with main to resolve file conflicts. Could you take a look at these new changes, and if nothing else is missing, merge this PR?

- https://docs-radiantlabs-co--36.org.readthedocs.build/projects/modeling-api/en/36/request_structure.html#water-heating-systems
- https://docs-radiantlabs-co--36.org.readthedocs.build/projects/modeling-api/en/36/request_structure.html#new-water-heating-system

**Pull request merged**